tso-server/src/main/java/org/apache/omid/tso/PersistenceProcessorHandler.java

```diff
@@ -117,30 +124,37 @@ private void commitSuicideIfNotMaster() {
         }
     }
 
-    private void sendReplies(Batch batch, long batchSequence) {
-
-        int i = 0;
-        while (i < batch.getNumEvents()) {
-            PersistEvent e = batch.get(i);
-            if (e.getType() == PersistEvent.Type.ABORT && e.isRetry()) {
-                retryProc.disambiguateRetryRequestHeuristically(e.getStartTimestamp(), e.getChannel(), e.getMonCtx());
-                PersistEvent tmp = batch.get(i);
-                //TODO: why assign it?
-                batch.set(i, batch.get(batch.getNumEvents() - 1));
-                batch.set(batch.getNumEvents()  - 1, tmp);
-                if (batch.getNumEvents()  == 1) {
-                    batch.clear();
-                    replyProcessor.manageResponsesBatch(batchSequence, null);
-                    return;
-                }
+    void filterClientRetriesAndSendBatchToReplyProcessor(Batch batch, long batchSequence) {
```

Comment:
Done. Now it reads like this:
       ....
          case TIMESTAMP:
                event.getMonCtx().timerStart("timestampPersistProcessor");
                break;
            }
        }

        // Flush and send the responses back to the client. WARNING: Before sending the responses, first we need
        // to filter commit retries in the batch to disambiguate them.
        flush(commitEventsToFlush);
        filterAndDissambiguateClientRetries(batch);
        replyProcessor.manageResponsesBatch(batchEvent.getBatchSequence(), batch);

    }